### PR TITLE
Make ATDM Trilinos configuration match EMPIRE Trilinos configuration

### PR DESCRIPTION
**Pull request by bartlettroscoe**
This PR updates the base ATDM Trilinos configuration to match the EMPIRE Trilinos configuration for the following settings:

* MueLu_ENABLE_Kokkos_Refactor_Use_By_Default=ON by default for CUDA builds (#3941)
* Set Trilinos_ENABLE_DEBUG=ON for all 'debug' ATDM Trilinos bulds (#4201)
* Remove Kokkos_ENABLE_Profiling=OFF from current ATDM Trilinos configuration? (#4148)

It turns out that I was not able to also completely address:

* Add option to turn off the install of gtest header and lib even if Gtest package is enabled (#5341)

but I made all of the changes to Trilinos to disable the install of Gtest without disabling the Gtest package itself.  It is just that SPARC currently requires Gtest to be installed (see [SPAR-614](https://sems-atlassian-srn.sandia.gov/browse/SPAR-614)).

I had to make some changes to the TriBITS CMake files to deal with the Gtest issue in commits 1836f97 and cad3fb5.  (I will create patches for these and apply them to STK in the Sierra.base repo.)

I also had to fix a small issue in TriBITS in commit  3f01e25 but I already committed that patch to TriBITS GitHub 'master' and snapshottted an updated TriBITS in commit 13e14bd and merged in updated 'tribits_github_snapshot' branch into this branch as per the workflow described in http://trac.trilinos.org/wiki/TriBITSTrilinosDev.

See other changes in other commits as well.

## How was this tested?

I ran several ATDM Trilinos builds and tested against SPARC and EMPIRE over several days in different stages of the commits that are described in detail in [ATDV-202](https://sems-atlassian-srn.sandia.gov/browse/ATDV-202).  The Trilinos builds were:

* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4854269) (passed=2071)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4854382) (passed=2021)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4854453) (passed=2026)
* [Trilinos-atdm-waterman-cuda-9.2-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4854338) (passed=2187, failed=5)
* [Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4854574) (passed=2199)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4856744) (passesd=2071)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4856850) (passed=2021)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4857008) (passed=2026)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4857034) (passed=2026)
* [Trilinos-atdm-sems-rhel6-intel-17.0.1-openmp-release-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4857060) (passed=2021)
* [Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4859452) (passed=2096)
* [Trilinos-atdm-cee-rhel6_gnu-7.2.0_openmpi-1.10.2_serial_shared_opt-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4859506) (passed=2098)
* [Trilinos-atdm-cee-rhel6_intel-17.0.1_intelmpi-5.1.2_serial_static_opt-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4859560) (passed=2095)
* [Trilinos-atdm-cee-rhel6_clang-5.0.1_openmpi-1.10.2_serial_static_opt-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4866361) (passed=2096)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4868912) (passed=2024)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4868938) (passed=2021)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4869080) (passed=2071)
* [Trilinos-atdm-sems-rhel6-intel-17.0.1-openmp-release-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4871509) (passed=2026)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4871483) (passed=2071)
* [Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-debug-exp](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&parentid=4871431) (passed=2021)

For the build `Trilinos-atdm-waterman-cuda-9.2-release-debug-exp` those 5 test failures were:

```
Subproject 	Name 	Status 	Time 	Proc Time 	Details 	Labels 	Summary 	Processors 
Intrepid2	Intrepid2_unit-test_Orientation_Cuda_Test_OrientationToolsModifyBasis_HCURL_MPI_1	Failed	4s 170ms	4s 170ms	Completed (Failed)	Intrepid2	Summary	1
Intrepid2	Intrepid2_unit-test_Orientation_Cuda_Test_OrientationToolsModifyBasis_HDIV_MPI_1	Failed	4s 170ms	4s 170ms	Completed (Failed)	Intrepid2	Summary	1
Intrepid2	Intrepid2_unit-test_Orientation_Cuda_Test_OrientationToolsModifyBasis_HGRAD_MPI_1	Failed	4s 690ms	4s 690ms	Completed (Failed)	Intrepid2	Summary	1
Intrepid2	Intrepid2_unit-test_Orientation_Cuda_Test_OrientationToolsModifyBasis_LowerOrder_MPI_1	Failed	4s 610ms	4s 610ms	Completed (Failed)	Intrepid2	Summary	1
Tempus	Tempus_DIRK_ASA_MPI_1	Failed	10m 50ms	10m 50ms	Completed (Timeout)	Tempus	Summary	1
```

The Intrepid2 failures all showed:

```
Intrepid2] Abort
[Intrepid2] Error in file /ascldap/users/rabartl/Trilinos.base/BUILDS/WATERMAN/CTEST_S/Trilinos-atdm-waterman-cuda-9.2-release-debug/SRC_AND_BUILD/Trilinos/packages/intrepid2/src/Orientation/Intrepid2_OrientationDef.hpp, line 97
            Test that evaluated to true: ( subCellVerts[0] == subCellVerts[1] )
            >>> ERROR (Intrepid::Orientation::getOrientation): Invalid subCellVerts, same vertex ids are repeated 
```

We can create issues for these once we merge these changes and they show up in CDash builds.

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
@alanw0,

FYI: I pushed the following commits to the Sierra.base 'master' branch mirroring these changes for STK:

```
3218aaa "STK: Don't install if Gtest_SKIP_INSTALL=TRUE (trilinos/Trilinos#5341)" <rabartl@sandia.gov> [Tue Sep 17 11:40:11 2019 -0600] (2 minutes ago)
M       stk/stk_ngp_test/stk_ngp_test/CMakeLists.txt
M       stk/stk_unit_test_utils/stk_unit_test_utils/CMakeLists.txt

db99ff2 "STK: Remove parent STK deps on all other SE packages (#5341)" <rabartl@sandia.gov> [Tue Sep 17 10:16:31 2019 -0600] (14 minutes ago)
M       stk/cmake/Dependencies.cmake
```

These changes should have zero chance of breaking Sierra's build of Trilinos using the TriBITS CMake build system (if Sierra is using that).  If you have any questions about these changes, please let me know.

**Comment by alanw0** [comment]
Thanks @bartlettroscoe those changes look fine.

File: cmake/std/atdm/ctest-s-local-test-driver.sh [hunk 1]
```diff
@@ -8,17 +8,50 @@
...
+  Trilinos/cmake/ctest/drivers/atdm/<system_name>/drivers/
+    ${ATDM_CONFIG_CTEST_S_BUILD_NAME_PREFIX}<build-name-keysi>.sh
+
+For example, ATDM_CONFIG_CTEST_S_BUILD_NAME_PREFIX=Trilinos-atdm-waterman- and
+<build-name-keysi>=cuda-9.2-release-debug gives the full build name
+'Trilinos-atdm-waterman-cuda-9.2-release-debug'.
+
+However, if env var:
+
+  ATDM_CHT_FULL_BUILD_NAME=1
```
Comment (rppawlo) [inline cmake/std/atdm/ctest-s-local-test-driver.sh]: For my knowledge, what does "CHT" stand for?

Comment (bartlettroscoe) [reply cmake/std/atdm/ctest-s-local-test-driver.sh]: @rppawlo,

FYI: Changed this var name to `ATDM_CTEST_S_USE_FULL_BULD_NAME` in PR #5981 (which you just approved :-))

Thanks for taking the time to find and report issues like this!

**Pull request merged**